nat-lab/tests/utils/vm/mac_vm_util.py

```diff
@@ -28,21 +30,22 @@ async def new_connection(ip: str = config.MAC_VM_IP) -> AsyncIterator[Connection
         compression_algs=None,
     )
 
-    async with asyncssh.connect(
-        ip,
-        username="root",
-        password="vagrant",  # NOTE: this is hardcoded password for transient vm existing only during the tests
-        known_hosts=None,
-        options=ssh_options,
-    ) as ssh_connection:
-        connection = SshConnection(ssh_connection, TargetOS.Mac)
+    if copy_binaries:
+        async with asyncssh.connect(
+            ip,
+            username="root",
+            password="vagrant",  # NOTE: this is hardcoded password for transient vm existing only during the tests
+            known_hosts=None,
+            options=ssh_options,
+        ) as ssh_connection:
+            connection = SshConnection(ssh_connection, TargetOS.Mac)
 
-        await _copy_binaries(ssh_connection, connection)
+            await _copy_binaries(ssh_connection, connection)
```

Comment:
What happens if `_copy_binaries()` fails the first time (like with those time outs)? Do we need to check for that, or would the whole test fail?

Reply:
I've changed a bit whole concept, but yeah, if copying binaries fails to one of the VMs - the pytest session fails.